packages/smooth_app/lib/pages/product/nutrition_container.dart

```diff
@@ -268,19 +282,35 @@ class NutritionContainer {
   }
 
   /// Returns the unit key according to [Nutriments] json map.
-  static String _getNutrimentsUnitKey(final String nutrientId) =>
+  static String getUnitKey(final String nutrientId) =>
       '${_fixNutrientId(nutrientId)}_unit';
 
   static String _fixNutrientId(final String nutrientId) =>
-      nutrientId == _energyKJId ? _energyId : nutrientId;
+      nutrientId == energyKJId ? energyId : nutrientId;
+
+  /// Converts a double (weight) value from grams.
+  ///
+  /// Typical use-case: after receiving a value from the BE.
+  static double? convertWeightFromG(final double? value, final Unit unit) {
+    if (value == null) {
+      return null;
+    }
+    if (unit == Unit.MILLI_G) {
+      return value * 1E3;
```

Comment:
Dart supports scientific notations thats nice to know 🎉 ,
must for some reason have overread this in the original imlementation